app/services/support_interface/applications_export.rb

```diff
@@ -0,0 +1,60 @@
+module SupportInterface
+  class ApplicationsExport
+    def applications
+      relevant_applications.map do |application_form|
+        output = {
+          'Reference code' => application_form.support_reference,
+          'Stage' => ProcessState.new(application_form).state,
+          'Signed up' => application_form.created_at,
+          'Submitted at' => application_form.submitted_at,
+          'Date last change' => application_form.updated_at,
+          'Course select' => application_form.application_choices.sort_by(&:updated_at).last&.updated_at,
+          'Qualifications' => application_form.application_qualifications.sort_by(&:updated_at).last&.updated_at,
+          'Work history' => application_form.application_work_experiences.sort_by(&:updated_at).last&.updated_at,
+          'References' => application_form.application_references.sort_by(&:updated_at).last&.updated_at,
+        }
+
+        %w[first_name english_main_language date_of_birth becoming_a_teacher subject_knowledge interview_preferences].each do |column|
+          output["#{column} completed"] = last_change_to_form(application_form, column)
+        end
+
+        output
+      end
+    end
+
+  private
+
+    def last_change_to_form(application_form, column)
+      audit = application_form.own_and_associated_audits.find do |audit|
+        audit.audited_changes.has_key?('start_date')
```

Comment:
```suggestion
        audit.audited_changes.has_key?(column)
```

Reply:
yes that makes sense now! Nice way to get a granular last updated at summary.

I guess it would be possible to also get a history of when the application reached each status in the workflow pretty much the same way. What I don't think we can do is get a history of 'actions'.

Reply:
Yeah, the actions would have to be inferred by looking at before/after `status`.